File: HashTable/mostCommonWord.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        banned = set(banned)
        dic = {}
        for c in "!?',;.":
            paragraph = paragraph.replace(c, " ")
        words = paragraph.lower().split(" ")
        for word in words:
            if word and word not in banned:
                if word not in dic:
                    dic[word] = 1
                else:
                    dic[word] += 1
        return max(dic, key = lambda key : dic[key])

File: HashTable/mostCommonWord.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> ban = new HashSet<>(Arrays.asList(banned));
        Map<String, Integer> count = new HashMap<>();
        String[] words = paragraph.replaceAll("\\W+", " ").toLowerCase().split("\\s+");
        for (String w : words) {
            if (!ban.contains(w)) 
                count.put(w, count.getOrDefault(w, 0) + 1);
        }
        int max = 0;
        String res = "";
        for (String w : count.keySet()) {
            if (count.get(w) > max) {
                max = count.get(w);
                res = w;
            }
        }
        return res;
    }
}
